pkg/services/folder/folderimpl/folder.go

```diff
@@ -464,27 +456,35 @@ func (s *Service) Update(ctx context.Context, cmd *folder.UpdateFolderCommand) (
 	}
 	user := cmd.SignedInUser
 
-	dashFolder, err := s.legacyUpdate(ctx, cmd)
-	if err != nil {
-		return nil, err
-	}
+	var dashFolder, foldr *folder.Folder
+	var err error
+	err = s.db.InTransaction(ctx, func(ctx context.Context) error {
+		if dashFolder, err = s.legacyUpdate(ctx, cmd); err != nil {
+			return err
+		}
 
-	foldr, err := s.store.Update(ctx, folder.UpdateFolderCommand{
-		UID:            cmd.UID,
-		OrgID:          cmd.OrgID,
-		NewTitle:       cmd.NewTitle,
-		NewDescription: cmd.NewDescription,
-		SignedInUser:   user,
-	})
-	if err != nil {
-		if errors.Is(err, folder.ErrFolderNotFound) {
-			logger.Warn("attempt to update folder that does not exist in the folders table", "folderUID", cmd.UID)
-			return dashFolder, nil
```

Comment:
Question about this: In my opinion, it is reasonable that if an update to the `dashboards` table succeeds but the update to the `folders` table fails, the entire transaction should be rolled back, and none of the changes should be committed. However, I would like to double-check whether this is the desired behavior.

Reply:
👍  sounds good to me